rust/lance-encoding/src/encoder.rs

```diff
@@ -291,6 +293,8 @@ impl CoreArrayEncodingStrategy {
         data_type: &DataType,
         data_size: u64,
         use_dict_encoding: bool,
+        use_fixed_size_encoding: bool,
+        byte_width: &u64,
```

Comment:
This feels like too many arguments to this method.  I'll comment an alternative later on.